core/src/main/scala/org/apache/spark/CacheManager.scala

```diff
@@ -44,7 +44,14 @@ private[spark] class CacheManager(blockManager: BlockManager) extends Logging {
     blockManager.get(key) match {
       case Some(blockResult) =>
         // Partition is already materialized, so just return its values
+        val existingMetrics = context.taskMetrics.inputMetrics
+        val prevBytesRead = existingMetrics
+          .filter(_.readMethod == blockResult.inputMetrics.readMethod)
+          .map(_.bytesRead)
+          .getOrElse(0L)
```

Comment:
Actually after looking at Hadoop RDD - it might be necessary to just clobber here to preserve consistency with that case. But it could still be nicer to write this with a match.